components/hdw-bzr/hdw-bzr.c

```diff
@@ -13,56 +15,82 @@
 // Defines
 //==============================================================================
 
-// For LEDC
 #define LEDC_MODE LEDC_LOW_SPEED_MODE //!< Low speed mode is sufficient
 
 //==============================================================================
-// Enums
+// Structs
 //==============================================================================
 
 /**
- * @brief A track for a song on the buzzer. This plays notes from a song_t
+ * @brief Types of loops for a track
  */
-typedef struct
+typedef enum
 {
-    const song_t* song;  ///< The song_t being played
-    uint32_t note_index; ///< The index of the current musicalNote_t in the song
-    int64_t start_time;  ///< The time the current musicalNote_t started in the song
-    uint16_t volume;     ///< This track's volume, from 0 (off) to 4096 (max)
-} buzzerTrack_t;
+    NO_LOOP,
+    MONO_LOOP,
+    STEREO_LOOP
+} loopType_t;
```

Comment:
What do these actually mean? Is it to keep track of whether both tracks loop together or separately?

Reply:
These should have been (and will be) removed. There was a moment where even though the time lengths of L and R tracks were identical, they would loop at slightly different times and desync. I put in a hack so that both tracks would loop whenever either finished.

It turned out the real problem was that `start_time` was being set to `cTime` rather than accumulating `noteTimeUs` (see a4a170d). This would lead to small drifts because notes are checked every 5ms, but note lengths do not need to be multiples of 5ms. Now the two tracks keep in sync, and the song is played at _exactly_ the right speed.

Reply:
Ahhh okay, that makes sense!